mergify_engine/merge_train.py

```diff
@@ -0,0 +1,510 @@
+        )
+        original_ctxt = context.Context(self.train.client, pull, sub, [])
+
+        title = "This pull request cannot tested to be merged"
+        summary = f"This pull request can't be tested to be merged because: {exception.message}"
+        original_ctxt.log.info(
+            "fail to create the pull request train car",
+            status=check_api.Conclusion.ACTION_REQUIRED,
+            title=title,
+            summary=summary,
+        )
+        check_api.set_check_run(
+            original_ctxt,
+            constants.MERGE_QUEUE_SUMMARY_NAME,
+            check_api.Result(
+                check_api.Conclusion.ACTION_REQUIRED,
+                title=title,
+                summary=summary,
+            ),
+        )
+
+        asyncio.run(github_events.send_refresh(original_ctxt.pull))
+
+    def update_summaries(
+        self,
+        tmp_pull_ctxt: context.Context,
+        queue_rule: typing.Optional[rules.EvaluatedQueueRule],
+        status: check_api.Conclusion,
+        deleted: bool,
+    ) -> None:
+        if status == check_api.Conclusion.SUCCESS:
+            title = f"The pull request #{self.pull_request_number} can be merged"
+        elif status == check_api.Conclusion.PENDING:
+            title = (
+                f"The pull request #{self.pull_request_number} is tested to be merged"
+            )
+        else:
+            title = f"The pull request #{self.pull_request_number} cannot be merged"
+
+        if queue_rule:
+            queue_summary = "\n\nRequired conditions for merge:\n"
+            for cond in queue_rule.conditions:
+                checked = " " if cond in queue_rule.missing_conditions else "X"
+                queue_summary += f"\n- [{checked}] `{cond}`"
+        else:
+            queue_summary = ""
+
+        pulls_for_title = ", ".join(
+            [f"#{p}" for p in self.parents + [self.pull_request_number]]
+        )
+        summary = (
+            f"Testing {self.initial_current_base_sha[:7]}, {pulls_for_title} together"
+        )
+        summary += queue_summary
+
+        asyncio.run(
+            tmp_pull_ctxt.set_summary_check(
+                check_api.Result(
+                    status,
+                    title=title,
+                    summary=summary,
+                )
+            )
+        )
+
+        # Update the original Pull Request
+        pull: github_types.GitHubPullRequest = self.train.client.item(
+            f"/repos/{self.train.owner_login}/{self.train.repo_name}/pulls/{self.pull_request_number}"
+        )
+
+        sub = asyncio.run(
+            subscription.Subscription.get_subscription(self.train.owner_id)
+        )
+        original_ctxt = context.Context(self.train.client, pull, sub, [])
+        title = f"The pull request is tested against {original_ctxt.pull['base']['ref']} to be merged"
+        if deleted:
+            # TODO(sileht): display train cars ?
+            summary = "The pull request is going to be tested soon"
```

Comment:
```suggestion
            summary = "The pull request is going to be queued soon"
```

Reply:
Rewording:

queued can be confusing as the PR is first queued in the Queue and then queued in the Train.

Reply:
Understood. In that case `queue` is not the right word indeed.
What about using `embarked`?
We could use `embark` and `disembark` as the verbs to described what happens to PRs that are in queue and now art part of a train.

Reply:
far better :smile: